File: Neetcode/1DDP/climbingstairs.cpp
#include <vector>
using namespace std;

class Solution {
 public:
  int climbStairs(int n) {
    vector<int> track(n + 1, 1);
    for (int i = 2; i < n + 1; i++) {
      track[i] = track[i - 2] + track[i - 1];
    }
    return track[n];
  }
};

File: Neetcode/1DDP/climbingstairs.py
# https://neetcode.io/problems/climbing-stairs
# https://leetcode.com/problems/climbing-stairs/


class Solution:
    def climbStairs(self, n: int) -> int:
        track = [1] * (n + 1)
        for i in range(2, n + 1):
            track[i] = track[i - 2] + track[i - 1]
        return track[-1]
